### PR TITLE
fix apply_set_last_applied dry-run output issue

### PR DESCRIPTION
**Pull request by shiywang**
Fixes https://github.com/kubernetes/kubernetes/issues/50572

@kubernetes/sig-cli-bugs 

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/apply_set_last_applied.go [hunk 1]
```diff
@@ -222,9 +222,9 @@ func (o *SetLastAppliedOptions) formatPrinter(output string, buf []byte, w io.Wr
 		if err != nil {
 			return err
 		}
-		fmt.Fprintf(w, string(jsonBuffer.Bytes()))
+		fmt.Fprintf(w, string(jsonBuffer.Bytes())+"\n")
```
Comment (mengqiy) [inline pkg/kubectl/cmd/apply_set_last_applied.go]: It's not safe if `jsonBuffer` has `%` in it.
I suggest change it to
```
fmt.Fprintf(w, "%s", jsonBuffer.String())
fmt.Fprintf(w, "\n")
```

**Comment by shiywang** [comment]
@mengqiy ptal

File: pkg/kubectl/cmd/apply_set_last_applied.go [hunk 2]
```diff
@@ -222,9 +222,11 @@ func (o *SetLastAppliedOptions) formatPrinter(output string, buf []byte, w io.Wr
 		if err != nil {
 			return err
 		}
-		fmt.Fprintf(w, string(jsonBuffer.Bytes()))
+		fmt.Fprintf(w, "%s", jsonBuffer.String())
```
Comment (thockin) [inline pkg/kubectl/cmd/apply_set_last_applied.go]: why not "%s\n" in both cases?

**Comment by shiywang** [comment]
@thockin done, ptal

**Comment by alexandercampbell** [comment]
@shiywang awesome, thanks for catching & fixing this.

**Pull request merged**